gcp-client/src/main/java/com/google/cloud/graphite/platforms/plugin/client/ContainerAnalysisClient.java

```diff
@@ -0,0 +1,189 @@
+        Comparator.comparing(Occurrence::getName));
+  }
+
+  /**
+   * @param projectId The project ID where the container image is hosted.
+   * @param resourceUrl The complete URL of the image in this form:
+   *     https://[HOST_NAME]/[PROJECT_ID]/[IMAGE_ID]@sha256:[HASH]
+   * @param noteProjectId The project ID of the note representing the attestation authority.
+   * @param noteId The note ID of the note representing the attestation authority.
+   * @param signature The signed payload created using the key referenced by the publicKeyId.
+   * @param publicKeyId The id of the public key for the asymmetric PIKIX key used to sign the
+   *     payload and produce the signature.
+   * @param payload The base64-encoded attestation verified by the provided signature.
+   * @return The attestation {@link Occurrence} that was created.
+   * @throws IOException An error occurred attempting to get the occurrence.
+   */
+  public Occurrence createAttestation(
+      final String projectId,
+      final String resourceUrl,
+      final String noteProjectId,
+      final String noteId,
+      final String signature,
+      final String publicKeyId,
+      final String payload)
+      throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(resourceUrl));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(noteProjectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(noteId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(signature));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(payload));
+    return containerAnalysis.createAttestation(
+        projectId, resourceUrl, noteProjectId, noteId, signature, publicKeyId, payload);
+  }
+
+  private Optional<Occurrence> getStatusOccurrence(String projectId, String resourceUrl) {
+    List<Occurrence> occurrences;
+    try {
+      occurrences =
+          containerAnalysis.listOccurrences(
+              projectId,
+              buildFilterString(
+                  ImmutableMap.of(
+                      "resourceUrl",
+                      resourceUrl,
+                      "noteProjectId",
+                      VULNERABILITY_NOTE_PROJECT_ID,
+                      "noteId",
+                      VULNERABILITY_NOTE_ID)));
+    } catch (IOException ioe) {
+      log.info(String.format("Error listing occurrences: %s. Retrying ...", ioe.getMessage()));
+      return Optional.empty();
+    }
+    for (Occurrence o : occurrences) {
+      if (o.getDiscovered() != null) {
+        return Optional.of(o);
+      }
+    }
+    log.info("Did not find occurrences.");
+    return Optional.empty();
+  }
+
+  private Optional<String> getVulnerabilityScanStatus(String projectId, String occurrenceId) {
+    Occurrence occurrence;
+    try {
+      occurrence = containerAnalysis.getOccurrence(projectId, occurrenceId);
+    } catch (IOException ioe) {
+      log.info(
```

Comment:
Suggest using a higher log level for errors.

Reply:
Done